src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -400,4 +419,104 @@ else if (e instanceof PasswordGuardrail.PasswordGuardrailException)
 
         return PasswordObfuscator.obfuscate(e.getMessage());
     }
+
+    private static class JmxFormatter
+    {
+        private static String user(Subject subject)
+        {
+            return String.format("%s", subject == null ? null : subject.getPrincipals().stream().map(Objects::toString).collect(Collectors.joining(", ")));
+        }
+
+        private static String method(Method method, Object[] args)
+        {
+            String argsFmt = "";
+            if (args != null)
+                argsFmt = Arrays.stream(args).map(Objects::toString).collect(Collectors.joining(", "));
+            return String.format("%s#%s(%s)", method.getDeclaringClass().getCanonicalName(), method.getName(), argsFmt);
+        }
+    }
+
+    @Override
+    public void onInvocation(Subject subject, Method method, Object[] args)
+    {
+        if (filter.isFiltered(AuditLogEntryCategory.JMX))
+            return;
+
+        AuditLogEntry entry = new AuditLogEntry.Builder(JMX)
+                              .setOperation(String.format("JMX INVOCATION: %s", JmxFormatter.method(method, args)))
+                              .setUser(JmxFormatter.user(subject))
+                              .build();
+        log(entry);
+    }
+
+    @Override
+    public void onFailure(Subject subject, Method method, Object[] args, String reason)
+    {
+        if (filter.isFiltered(AuditLogEntryCategory.JMX))
+            return;
+
+        AuditLogEntry entry = new AuditLogEntry.Builder(JMX)
+                              .setOperation(String.format("JMX FAILURE: %s due to %s", JmxFormatter.method(method, args), reason))
+                              .setUser(JmxFormatter.user(subject))
+                              .build();
+        log(entry);
+    }
+
+
+    private class JmxHandler implements InvocationHandler
+    {
+        private MBeanServer mbs = null;
+
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args) throws Throwable
+        {
+            // See AuthorizationProxy.invoke
+            if (("setMBeanServer").equals(method.getName()))
```

Comment:
nit: remove redundant unnecessary parentheses
```suggestion
            if ("setMBeanServer".equals(method.getName()))
```